earthaccess/services.py

```diff
@@ -0,0 +1,73 @@
+from typing import Any, List, Optional
+
+from requests import exceptions, session
+
+from cmr import ServiceQuery
+
+from .auth import Auth
+
+
+class DataService(ServiceQuery):
```

Comment:
Any reason this is not in `search.py`, like `DataCollections` and `DataGranules` are?

I'm not opposed to keeping this in a separate file, but splitting each type of query class into their own modules might perhaps be left as a task for a separate issue (after discussing whether or not we want to do so).

For consistency with existing code, I suggest moving this to `search.py` and also renaming it to be pluralized: `DataServices`.

Reply:
I had initially wanted to place `DataService` in `search.py` but because the results module uses the `DataService` class to query and parse results, a circular dependency is created between the search and results modules. I can look into re-architecting the code so that there is a `DataServices` class in `search.py` and `DataService` class in `results.py` to be more consistent with the Collections and Granules structure. I was initially thinking that the services would be returned for a collection rather than having a separate `search_services` function. Open to suggestions and/or discussing at the next hackday.

Reply:
Ah, I see the circular dependency you're referring to.

I still suggest your rename `DataService` (singular) to `DataServices` (plural) making it consistent with `DataCollections` and `DataGranules`.

Regarding the "easier" code you mention above, I agree, but that doesn't preclude providing a `search_services` method as well. Users can then do both things: (a) call `dataset.services()` to get the services associated with a dataset, or (b) call `search_services` to search for services more generally, not necessarily specific to a dataset.

Reply:
>Users can then do both things: (a) call dataset.services() to get the services associated with a dataset, or (b) call search_services to search for services more generally, not necessarily specific to a dataset.

Should we consider these separate features and follow-up later to add even more convenience?

Reply:
I don't think so. In fact, I suggest that `dataset.services()` simply invoke `search_services`.

Reply:
Thanks for the feedback @chuckwondo - I think I was trying to keep the scope of the work for this issue to a manageable size so we could also start to work in the plugin architecture. We can certainly add a `search_services` method. I also anticipated maybe wanting to find all services available for granules. These could each be their own GitHub issue.

I am leaning towards keeping things as is, rename `DataService` to `DataServices` and extract `earthaccess.search.get_results` out to a separate utilities module. Maybe: `earthaccess.utils._search.py` and then refactoring its usage in `search.py`. Then the `services` function in `results.py` can import without the circular dependency. This way we can still support the "easier" code and build off the DataServices class for the plugin architecture.

But I will mull things over a little bit more and most likely tackle this at the hackday next week if you want to tag up to discuss further.

Reply:
Sounds reasonable to me. My only request is to _not_ name anything "utils." It's a pet peeve of mine because it's such a generic name, as to have no meaning. Happy to iron out kinks with you at the next hack day.

Reply:
>My only request is to not name anything "utils."

I love this and feel so called out :rofl: I'm very prone to creating a `utils` subpackage but I always regret it later and am trying to get better at it.

Reply:
Totally agree on `utils` 😄  but it does look like there is already a `utils` [directory](https://github.com/nsidc/earthaccess/tree/main/earthaccess/utils). Should we consider moving that to a different name? Or maybe I am misunderstanding!

Reply:
Oh, I hadn't noticed that there's already a `utils`. Oh well. We can worry about that another time.

Reply:
If that was me, sorry :rofl: